theoriq/api/v1alpha2/protocol/protocol_client.py

```diff
@@ -0,0 +1,121 @@
+from __future__ import annotations
+
+import os
+from collections import OrderedDict
+from datetime import datetime, timezone
+from typing import Any, Dict, List, Optional, Sequence
+
+import httpx
+
+from theoriq.biscuit import AgentAddress, RequestBiscuit
+from theoriq.types import Metric
+from theoriq.utils import is_protocol_secured
+
+from ..schemas.agent import AgentResponse
+from ..schemas.api import PublicKeyResponse
+from ..schemas.event_request import EventRequestBody
+from ..schemas.metrics import MetricsRequestBody
+
+
+class ProtocolClient:
+    _config_cache: OrderedDict[str, Dict[str, Any]] = OrderedDict()
+
+    def __init__(self, uri: str, timeout: Optional[int] = 120, max_retries: Optional[int] = None):
+        self._uri = f"{uri}/api/v1alpha2"
+        self._timeout = timeout
+        self._max_retries = max_retries or 0
+        self._public_key: Optional[str] = None
+
+    @property
+    def public_key(self) -> str:
+        if self._public_key is None:
+            self._public_key = self.get_public_key().public_key
+        return self._public_key
+
+    def get_public_key(self) -> PublicKeyResponse:
+        with httpx.Client(timeout=self._timeout) as client:
+            response = client.get(url=f"{self._uri}/auth/biscuits/public-key")
+            response.raise_for_status()
+            data = response.json()
+            return PublicKeyResponse(**data)
+
+    def get_agents(self) -> Sequence[AgentResponse]:
+        with httpx.Client(timeout=self._timeout) as client:
+            response = client.get(url=f"{self._uri}/agents")
+            response.raise_for_status()
+            data = response.json()
+            return [AgentResponse(**item) for item in data["items"]]
+
+    def get_configuration(self, request_biscuit: RequestBiscuit, agent_address: AgentAddress) -> Dict[str, Any]:
+        key = agent_address.address
```

Comment:
Using only the `agent_address` as a key cache would return a cache if the configuration for that agent has changed.
To prevent this, the configuration hash might need to be used instead, or in addition